frontend/flaskserver.py

```diff
@@ -0,0 +1,115 @@
+import flask
+import traceback
+import os
+import sys
+import pymysql.cursors
+import subprocess
+import datetime
+import imp
+
+
+# alabno/frontend
+exec_dir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)))
+
+pdf_path_dir = os.path.abspath(exec_dir + '/..')
+
+upload_path_dir = os.path.abspath(exec_dir + '/../uploads')
+subprocess.call('mkdir {}'.format(upload_path_dir), shell=True)
+
+# import mysqldb.py
+mysqldb_dir = exec_dir + os.sep + '..' + os.sep + 'simple-haskell-marker'
+sys.path.append(mysqldb_dir)
+import mysqldb
+
+
+def datetime_past_seconds(time):
+    now = datetime.datetime.now()
+    difference = now - time
+    return difference.total_seconds()
+    
+
+def get_pdf_path(db, token):
+    sql = 'SELECT path, created FROM PdfPaths WHERE token=%s'
+    results = db.query(sql, [token])
+    if results is None:
+        print('No path given token {} found'.format(token))
+        return None
+    try:
+        final_path = results[0]['path']
+        final_timestamp = results[0]['created']
+        seconds_past = datetime_past_seconds(final_timestamp)
+        if seconds_past > (60 * 20): # expiration is after 1 hour
```

Comment:
Idk why it outdated my earlier comment  but I wrote - For 1 hour wouldn't it be 60 * 60?